src/main/java/org/cryptomator/common/locationpresets/GoogleDriveMacLocationPresetsProvider.java

```diff
@@ -0,0 +1,109 @@
+public final class GoogleDriveMacLocationPresetsProvider implements LocationPresetsProvider {
+	private static final Path ROOT_LOCATION = LocationPresetsProvider.resolveLocation("~/Library/CloudStorage/").toAbsolutePath();
+	private static final Predicate<String> PATTERN = Pattern.compile("^GoogleDrive-[^/]+$").asMatchPredicate();
+
+	private static final List<Path> FALLBACK_LOCATIONS = Arrays.asList( //
+			LocationPresetsProvider.resolveLocation("~/GoogleDrive/My Drive"), //
+			LocationPresetsProvider.resolveLocation("~/Google Drive/My Drive"), //
+			LocationPresetsProvider.resolveLocation("~/GoogleDrive"), //
+			LocationPresetsProvider.resolveLocation("~/Google Drive") //
+	);
+
+	@Override
+	public Stream<LocationPreset> getLocations() {
+		if(isRootLocationPresent()) {
+			return getCloudStorageDirLocations();
+		} else {
+			return getFallbackLocation();
+		}
+	}
+
+	@CheckAvailability
+	public static boolean isPresent() {
+		return isRootLocationPresent() || FALLBACK_LOCATIONS.stream().anyMatch(Files::isDirectory);
+	}
+
+	public static boolean isRootLocationPresent() {
+		try (DirectoryStream<Path> stream = Files.newDirectoryStream(ROOT_LOCATION, "GoogleDrive-*")) {
+			return StreamSupport.stream(stream.spliterator(), false).anyMatch(Files::isDirectory);
+		} catch (IOException | UncheckedIOException e) {
+			return false;
+		}
+	}
+
+	/**
+	 * Returns Google Drive preset String.
+	 *
+	 * @param accountPath The path to the Google Drive account directory (e.g. {@code ~/Library/CloudStorage/GoogleDrive-username})
+	 * @param drivePath The path to the Google Drive file directory, within the account directory. (e.g. {@code ~/Library/CloudStorage/GoogleDrive-username/drive_name})
+	 * @return {@code String}. For example: "Google Drive - username - drive_name"
+	 */
+	private String getDriveLocationString(Path accountPath, Path drivePath) {
+		String accountName = accountPath.getFileName().toString().replace("GoogleDrive-", "");
+		String driveName = drivePath.getFileName().toString();
+
+		return STR."Google Drive - \{accountName} - \{driveName}";
+	}
+
+	private Stream<LocationPreset> getCloudStorageDirLocations() {
+		try (var dirStream = Files.list(ROOT_LOCATION)) {
+			return dirStream.filter(path -> Files.isDirectory(path) && PATTERN.test(path.getFileName().toString()))
+					.flatMap(this::getPresetsFromAccountPath)
+					.filter(preset -> Files.isWritable(preset.path()))
+					.toList().stream();
+		} catch (IOException | UncheckedIOException e) {
+			return Stream.empty();
+		}
+	}
+
+	private Stream<LocationPreset> getPresetsFromAccountPath(Path accountPath) {
+		try (var driveStream = Files.newDirectoryStream(accountPath, Files::isDirectory)) {
+			List<Path> directories = StreamSupport.stream(driveStream.spliterator(), false).toList();
+			return directories.stream()
+					.filter(drivePath -> {
+						try {
+							return !Files.isHidden(drivePath);
+						} catch (IOException e) {
+							throw new RuntimeException(e);
```

Comment:
This is bad exception to throw. Not only is it catched nowwhere, additionally, why throw the exception at all? Our goal is to list all not hidden directories. If this cannot be determined for a directory, somethings broken there and we should just skip it. We can log the exception on debug level, but otherwise just swallow it.